cmd/hack/hack.go

```diff
@@ -1666,6 +1681,250 @@ func GenerateTxLookups1(chaindata string, block int) {
+			}
+		}
+		blockNum++
+		if blockNum%100000 == 0 {
+			log.Info("Processed", "blocks", blockNum, "iterations", iterations)
+			var m runtime.MemStats
+			runtime.ReadMemStats(&m)
+			log.Info("Memory", "alloc", int(m.Alloc/1024), "sys", int(m.Sys/1024), "numGC", int(m.NumGC))
+		}
+	}
+	batch := db.NewBatch()
+	h := &Heap{}
+	heap.Init(h)
+	for i := range lookups {
+		file, err := os.Open(fmt.Sprintf(".lookups_%d.tmp", i))
+		check(err)
+		lookups[i].file = file
+		lookups[i].reader = bufio.NewReader(file)
+		lookups[i].buffer = make([]byte, bufferLen)
+		check(err)
+		n, err := lookups[i].file.Read(lookups[i].buffer)
+		if n != bufferLen {
+			lookups[i].buffer = lookups[i].buffer[:n]
+		}
+		heap.Push(h, HeapElem{lookups[i].buffer[:35], i})
+	}
+
+	for !interrupt && len(*h) != 0 {
+		val := (heap.Pop(h)).(HeapElem)
+		if lookups[val.index].pos == uint64(bufferLen) {
+			if val.val[32] != 0 {
+				err := batch.Put(dbutils.TxLookupPrefix, val.val[:32], common.CopyBytes(val.val[32:]))
+				check(err)
+			} else {
+				err := batch.Put(dbutils.TxLookupPrefix, val.val[:32], common.CopyBytes(val.val[33:]))
+				check(err)
+			}
+			n, _ := lookups[val.index].reader.Read(lookups[val.index].buffer)
+			iterations++
+			if n == 0 {
+				err := lookups[val.index].file.Close()
+				check(err)
+				os.Remove(fmt.Sprintf(".lookups_%d.tmp", val.index))
+			} else {
+				if n != bufferLen {
+					lookups[val.index].buffer = lookups[val.index].buffer[:n]
+				}
+				lookups[val.index].pos = 35
+				heap.Push(h, HeapElem{lookups[val.index].buffer[:35], val.index})
+			}
+			continue
+		}
+
+		heap.Push(h, HeapElem{lookups[val.index].buffer[lookups[val.index].pos : lookups[val.index].pos+35], val.index})
+		lookups[val.index].pos += 35
+		iterations++
+		if val.val[32] != 0 {
+			err := batch.Put(dbutils.TxLookupPrefix, val.val[:32], common.CopyBytes(val.val[32:]))
+			check(err)
+		} else {
+			err := batch.Put(dbutils.TxLookupPrefix, val.val[:32], common.CopyBytes(val.val[33:]))
+			check(err)
+		}
+
+		if iterations%1000000 == 0 {
+			batch.Commit()
+			log.Info("Commit Occured", "progress", iterations)
+		}
+		select {
+		case interrupt = <-interruptCh:
+			log.Info("interrupted, please wait for cleanup...")
+		default:
+		}
+	}
+	batch.Commit()
+	log.Info("Process done", "duration", time.Since(startTime))
+	log.Info("Validation Start")
+
+	blockNum = startBlock
+	iterations = 0
+	// Validation Process
+	blockBytes := big.NewInt(0)
+	for !interrupt {
+		blockHash := rawdb.ReadCanonicalHash(db, blockNum)
+		body := rawdb.ReadBody(db, blockHash, blockNum)
+
+		if body == nil {
+			break
+		}
+
+		select {
+		case interrupt = <-interruptCh:
+			log.Info("interrupted, please wait for cleanup...")
+		default:
+		}
+		blockBytes.SetUint64(blockNum)
+		bn = blockBytes.Bytes()
+
+		for _, tx := range body.Transactions {
+			val, err := db.Get(dbutils.TxLookupPrefix, tx.Hash().Bytes())
+			iterations++
+			if iterations%100000 == 0 {
+				batch.Commit()
```

Comment:
You probably don't want this commit here. It slows down the validation, I suspect